webpack/scenes/RedHatRepositories/components/RepositoryTypeIcon.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { ListView, OverlayTrigger, Tooltip } from 'patternfly-react';
+
+import { getTypeIcon } from '../../../services/index';
+
+export default class RepositoryTypeIcon extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.tooltipId = `type-tooltip-${props.id}`;
+  }
+
+  render() {
+    return (
+      <OverlayTrigger
+        overlay={<Tooltip id={this.tooltipId}>{this.props.type}</Tooltip>}
+        placement="bottom"
+        trigger={['hover', 'focus']}
+        rootClose={false}
+      >
+        <ListView.Icon name={getTypeIcon(this.props.type)} />
+      </OverlayTrigger>
+    );
+  }
+}
+
+RepositoryTypeIcon.propTypes = {
+  id: PropTypes.number.isRequired,
+  type: PropTypes.string.isRequired,
```

Comment:
I'm getting a prop type warning on repos load:
`Warning: Failed prop type: Invalid prop 'id' of type 'string' supplied to 'RepositoryTypeIcon', expected 'number'.`
Not sure if it's better to cast it to number or change the prop type.

Reply:
Nice catch.  It should be a string, updated.

Reply:
Now I'm getting the opposite error:
```
Failed prop type: Invalid prop `id` of type `number` supplied to `RepositoryTypeIcon`, expected `string`.
```
I guess that you really need to cast it in the reducer.

Reply:
I ended up fixing this in the API so that the API will now return an integer instead of a string.

Reply:
Never mind, this is now fixed client-side in the reducers.